hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -151,7 +154,7 @@ object HoodieSparkSqlWriter {
           .setBaseFileFormat(baseFileFormat)
           .setArchiveLogFolder(archiveLogFolder)
           .setPayloadClassName(hoodieConfig.getString(PAYLOAD_CLASS_NAME))
-          .setPreCombineField(hoodieConfig.getStringOrDefault(PRECOMBINE_FIELD, null))
+          .setPreCombineField(optParams.getOrElse(PRECOMBINE_FIELD.key(), null))
```

Comment:
if not for this fix, tests written in TestMORDatasource fails

Reply:
Yeah, i've created a ticket to revisit the way we handle defaults a while back, but for now we have to come up with remediations like that

Reply:
Please add a comment elaborating why this is channelled from `optParams` and not `hoodieConfig` like everything else